common/thorhelper/thorsoapcall.cpp

```diff
@@ -1242,7 +1242,10 @@ class CWSCHelper : implements IWSCHelper, public CInterface
         if (!ownedSC)
         {
             if (clientCert != NULL)
-                ownedSC.setown(createSecureSocketContextEx(clientCert->certificate, clientCert->privateKey, clientCert->passphrase, ClientSocket));
+            {
+                Owned<IPropertyTree> config = createSecureSocketConfig(clientCert->certificate, clientCert->privateKey, clientCert->passphrase);
+                ownedSC.setown(createSecureSocketContextEx2(config, ClientSocket));
+            }
             else if (clientCertIssuer.length())
                 ownedSC.setown(createSecureSocketContextSecret(clientCertIssuer.str(), ClientSocket));
```

Comment:
@afishbeck can you clarify this?  I think there is a problem when (certificate, key, passphrase are provided) - and I have created a jira for that.
Saving a context created from a secret globally should not matter though - because it will get updated.

Reply:
@ghalliday Yes, sorry I looked at the wrong thing.  This should be fine.